staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -845,32 +757,9 @@ type listResp struct {
 }
 
 // GetList implements storage.Interface
-func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error {
+func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object, listRV uint64) error {
 	recursive := opts.Recursive
-	resourceVersion := opts.ResourceVersion
 	pred := opts.Predicate
-	if shouldDelegateList(opts) {
-		return c.storage.GetList(ctx, key, opts, listObj)
-	}
-
-	listRV, err := c.versioner.ParseResourceVersion(resourceVersion)
-	if err != nil {
-		return err
-	}
-
-	if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
-		if !c.ready.check() && shouldDelegateListOnNotReadyCache(opts) {
```

Comment:
It seems this is a behavior change. 
After this change, now it always reject. [Code](https://github.com/kubernetes/kubernetes/blob/4a4fc9da801f299176c7200e66224aa79b1c0817/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L777-L782)
The behavior is different what the KEP described in https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4568-resilient-watchcache-initialization#adjust-what-lists-are-delegated-to-etcd
@serathius  Is this desired?

Reply:
There is not behavior change. You are not looking at the correct code.

Reply:
Got it. It sounds like it now go through code path of `CacheProxy.GetList` instead of `Cacher.GetList`.